pkg/services/sqlstore/user_auth.go

```diff
@@ -101,8 +101,21 @@ func GetUserByAuthInfo(query *models.GetUserByAuthInfoQuery) error {
 		return models.ErrUserNotFound
 	}
 
+	// Special case for generic oauth duplicates
+	if query.AuthModule == "oauth_generic_oauth" && user.Id != 0 {
```

Comment:
@marefr I think this fixes it - tested and seems to not add a new entry.

However, my concern is, what happens when you have same email and login name from two different oauth sources? In that case, a different entry is added (as it should be and we would expect) but they share a `user_id`, which I think is wrong? Shouldn't they have different `user_id` since they use different oauth providers?

Reply:
> Shouldn't they have different user_id since they use different oauth providers?

It depends. But the user_auth table was added from the beginning to keep external auth providers in sync with Grafana users.

Two different oauth sources will create 2 rows in the auth user_auth table because login with non-generic oauth will add an `auth_id` value which is the unique identifier from third-party auth provider. For generic oauth providing the unique identifier from third-party auth provider is not supported why this empty in user_auth table. Does this make sense?